app/javascript/packs/foremCreatorSignup.js

```diff
@@ -0,0 +1,110 @@
+function setDefaultUsername(event) {
+  if (
+    document
+      .getElementsByClassName('js-creator-signup-username-row')[0]
+      .classList.contains('hidden')
+  ) {
+    const name = event.target.value;
+    // It's the first user and so we can assume that this username is not taken.
+    const usernameHint = createUsernameHint(name);
+    setUsernameHint(usernameHint);
+    setUsernameField(usernameHint);
+    showHintRow();
+  }
+}
+
+function createUsernameHint(name) {
+  return name
+    .toLowerCase()
+    .replace(/[^a-zA-Z0-9]/g, '_')
+    .substr(0, 30);
+}
+
+function showHintRow() {
+  const hintRow = document.getElementsByClassName(
+    'js-creator-signup-username-hint-row',
+  )[0];
+  hintRow.classList.remove('hidden');
+}
+
+function setUsernameHint(usernameHint) {
+  const usernameHintDisplay = document.getElementsByClassName(
+    'js-creator-signup-username-hint',
+  )[0];
+  usernameHintDisplay.innerHTML = usernameHint;
+}
+
+function setUsernameField(usernameHint) {
+  const usernameField = document.getElementsByClassName(
+    'js-creator-signup-username',
+  )[0];
+  usernameField.value = usernameHint;
+}
+
+function showUsernameField() {
+  const usernameRow = document.getElementsByClassName(
+    'js-creator-signup-username-row',
+  )[0];
+  usernameRow.classList.remove('hidden');
+  focusUsernameInput(usernameRow);
+  hideHintRow();
+}
+
+function focusUsernameInput(usernameRow) {
+  // A timer with a count of 0 will run when the thread becomes idle
+  window.setTimeout(() => {
+    usernameRow.getElementsByTagName('input')[0].focus();
+  }, 0);
+}
+
+function hideHintRow() {
+  const hintRow = document.getElementsByClassName(
+    'js-creator-signup-username-hint-row',
+  )[0];
+  hintRow.classList.add('hidden');
+}
+
+function togglePasswordMask(event) {
+  event.preventDefault();
+  visible = !visible;
+  toggleAriaPressed(visible);
+  toggleAriaLabel(visible);
+  togglePasswordType(visible);
+  toggleEyeIcons(visible);
+}
+
+function toggleAriaPressed(visible) {
+  visibility.setAttribute('aria-pressed', visible);
+}
+
+function toggleAriaLabel(visible) {
+  const action = visible ? 'Hide' : 'Show';
+  visibility.setAttribute('aria-label', `${action} password`);
+}
```

Comment:
For a toggle button it's actually preferable in this case to leave the aria label always as "Show password" - there's a really good explanation of why [in this article](https://inclusive-components.design/toggle-button), but the TL;DR is that because the screen reader conveys the "pressed" state, the label doesn't need to.

For example - keeping the button with the same label means the two ways it will be announced to screen reader users are:

- Show password (unselected) 
- Show password (selected)

If we change the label the second state is announced as:

- Hide password (selected) 

This can make it a bit unclear - a user might not know if 'selected' in this context means it's hidden already 🙃  By deferring all of the communication about current state to `aria-pressed` we keep it really predictable and easier to understand

Reply:
Makes sense, thanks for the clear explanation, will remove it!

Reply:
✅